packages/scenes/src/behaviors/CursorSync.ts

```diff
@@ -0,0 +1,77 @@
+import { BusEvent, BusEventHandler, BusEventType, EventBus, EventFilterOptions } from '@grafana/data';
+import { DashboardCursorSync } from '@grafana/schema';
+import { filter, Observable, Unsubscribable } from 'rxjs';
+import { sceneGraph } from '../core/sceneGraph';
+import { SceneObjectBase } from '../core/SceneObjectBase';
+import { SceneObject, SceneObjectState } from '../core/types';
+
+interface CursorSyncState extends SceneObjectState {
+  sync: DashboardCursorSync;
+}
+
+/**
+ * This behavior will provide a cursor sync context within a scene.
+ */
+
+export class CursorSync extends SceneObjectBase<CursorSyncState> {
+  public constructor(state: Partial<CursorSyncState>) {
+    super({
+      ...state,
+      sync: state.sync || DashboardCursorSync.Off,
+    });
+  }
+
+  public getEventsBus = (panel: SceneObject) => {
+    if (!this.parent) {
+      throw new Error('EnableCursorSync cannot be used as a standalone scene object');
+    }
+    // Since EnableCursorSync is a behavior, it is not a parent to any object in the scene graph.
+    // We need to get it's parent in order to provide correct EventBus context to the children.
+    return new PanelContextEventBus(this.parent, panel);
+  };
+
+  public getEventsScope = () => {
+    if (!this.parent) {
+      throw new Error('EnableCursorSync cannot be used as a standalone scene object');
+    }
+    // Since EnableCursorSync is a behavior, it is not a parent to any object in the scene graph.
+    // We need to get it's parent in order to provide correct EventBus context to the children.
+    return this.parent.state.key;
+  };
+}
+
+// This serves as a shared EventsBus that is shared by children or CursorSync behavior.
+class PanelContextEventBus implements EventBus {
+  public constructor(private _source: SceneObject, private _eventsOrigin: SceneObject) {}
+
+  public publish<T extends BusEvent>(event: T): void {
+    (event as any).origin = this._eventsOrigin;
+    this._eventsOrigin.publishEvent(event, true);
+  }
+
+  public getStream<T extends BusEvent>(eventType: BusEventType<T>): Observable<T> {
+    return new Observable<T>((observer) => {
+      const handler = (event: T) => {
+        observer.next(event);
+      };
+
+      const sub = this._source.subscribeToEvent(eventType, handler);
+
+      return () => sub.unsubscribe();
+    }).pipe(filter((event) => (event as any).origin !== this._eventsOrigin));
```

Comment:
```suggestion
    return new Observable<T>((observer) => {
      const handler = (event: T) => {
         if (event.origin !== this) {
           observer.next(event);
         } 
      };

      const sub = this._source.subscribeToEvent(eventType, handler);

      return () => sub.unsubscribe();
    });
```

less rxjs the better :)

Reply:
Yes, but this assumes the origin is events bus - no filtering it here means that the comparison in GraphNG https://github.com/grafana/grafana/blob/panel-context-events-scope/packages/grafana-ui/src/components/GraphNG/GraphNG.tsx#L181 will fail.